### PR TITLE
[IIIF-1050] Add support for clearing req stmt and viewing dir to v3

### PR DESCRIPTION
**Pull request by ksclarke**
* Add clearRequiredStatement for appropriate classes
* Add clearViewingDirection for appropriate classes
* Add the viewing direction methods to Collection, which seems to have been missed before
* Lots of Eclipse auto-correct/reformatting things with the new Eclipse version :-(
* Add tests for clear* methods

I think probably the things which aren't required would be good to be wrapped in Optional(s), but that's a breaking change for another time.

### CODE REVIEW

File: src/main/java/info/freelibrary/iiif/presentation/v3/AccompanyingCanvas.java [hunk 1]
```diff
@@ -360,27 +362,25 @@ public final AccompanyingCanvas supplementWith(final Minter aMinter, final Strin
...
-            final AnnotationPage<SupplementingAnnotation>... aPageArray) {
+    public final AccompanyingCanvas addSupplementingPages(final AnnotationPage<SupplementingAnnotation>... aPageArray) {
         return (AccompanyingCanvas) super.addSupplementingPages(aPageArray);
     }
 
     @Override
-    public final AccompanyingCanvas addSupplementingPages(
-            final List<AnnotationPage<SupplementingAnnotation>> aPageList) {
+    public final AccompanyingCanvas
+            addSupplementingPages(final List<AnnotationPage<SupplementingAnnotation>> aPageList) {
```
Comment (markmatney) [inline src/main/java/info/freelibrary/iiif/presentation/v3/AccompanyingCanvas.java:371]: unable to reproduce this auto-formatting

File: src/test/java/info/freelibrary/iiif/presentation/v3/AnnotationCollectionTest.java [hunk 2]
```diff
@@ -43,11 +45,23 @@ public void testConstructorUriIdLabel() {
      */
     @Test
     public void testGetSetViewingDirection() {
-        final AnnotationCollection annotationCollection = new AnnotationCollection(ID.toString(), LABEL)
-                .setViewingDirection(ViewingDirection.LEFT_TO_RIGHT);
+        final AnnotationCollection annotationCollection =
+                new AnnotationCollection(ID.toString(), LABEL).setViewingDirection(ViewingDirection.LEFT_TO_RIGHT);
```
Comment (markmatney) [inline src/test/java/info/freelibrary/iiif/presentation/v3/AnnotationCollectionTest.java:49]: unable to reproduce this auto-formatting

**Review by markmatney — APPROVED** [review]
Looks good to me. I noted a couple places where I noticed that my Eclipse is still behaving differently, so I guess maybe our "formatting profile" is different? I'd like to match mine at some point.

File: src/test/java/info/freelibrary/iiif/presentation/v3/properties/StartTest.java [hunk 3]
```diff
@@ -44,8 +48,9 @@ public final void testSpecificResourceStart() {
         final String idURL = myLorem.getUrl();
         final String sourceURL = myLorem.getUrl();
 
-        json.put(Constants.TYPE, ResourceTypes.SPECIFIC_RESOURCE).put(Constants.ID, idURL).put(Constants.SOURCE,
-                sourceURL).put(Constants.SELECTOR, new JsonObject().put(Constants.TYPE, selector.getType()));
+        json.put(Constants.TYPE, ResourceTypes.SPECIFIC_RESOURCE).put(Constants.ID, idURL)
+                .put(Constants.SOURCE, sourceURL)
+                .put(Constants.SELECTOR, new JsonObject().put(Constants.TYPE, selector.getType()));
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/test/java/info/freelibrary/iiif/presentation/v3/properties/StartTest.java:53]: Hmm, okay. This update did update the freelib-build-tools configs. Does your eclipse automatically use those? But this particular reformatting isn't something that I think is controlled by the build tools configs. I was surprised by it, but I do kind of like it. Yes, let's compare again soon.

File: src/main/java/info/freelibrary/iiif/presentation/v3/AccompanyingCanvas.java (hunk 1 above)
Comment (ksclarke) [reply src/main/java/info/freelibrary/iiif/presentation/v3/AccompanyingCanvas.java:371]: Hmm...

File: src/test/java/info/freelibrary/iiif/presentation/v3/properties/StartTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (markmatney) [reply src/test/java/info/freelibrary/iiif/presentation/v3/properties/StartTest.java:53]: yes this particular formatting is wonderful imo

[the remark this answers is not in the document]
Comment (markmatney) [reply src/test/java/info/freelibrary/iiif/presentation/v3/properties/StartTest.java:53]: ah, I wasn't using the updated configs. I've updated them and am now able to reproduce :+1:

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]

**Pull request merged**